# [LAB-05] [PBT] 데이터 품질 점검

## 2. Diamond Price Prediction

### #01 준비작업

1. 라이브러리 참조

In [1]:
from jussam import load_data
from helpers import my_qtcheck

📦 연세대학교 주영아 교수가 제작한 라이브러리를 사용중입니다.
📧 Email(1): j.purplerose@yonsei.ac.kr
📧 Email(2): j.purplerose@gmail.com
📝 Website: https://juyounga.kr/
🔖 Version: 0.5.21


2. 데이터 불러오기

In [2]:
origin = load_data("diamonds")
origin.head()

📚 다이아몬드를 커팅, 색상, 투명도, 가격 및 기타 속성 조사한 데이터 셋 (출처: https://www.kaggle.com/datasets/shivam2503/diamonds)

    field    description
--  -------  -------------------------------------------------------------------
 0  price    다이아몬드 가격 (USD, $326 ~ $18,823)
 1  carat    중량 (0.2~5.01)
 2  cut      컷 품질 (Fair, Good, Very Good, Premium, Ideal)
 3  color    색상 등급 - J (worst) to D (best)
 4  clarity  투명도 등급 (I1 (worst), SI2, SI1, VS2, VS1, VVS2, VVS1, IF (best))
 5  x        길이 mm (0~10.74)
 6  y        너비 mm (0--58.9)
 7  z        두께 mm (0--31.8)
 8  depth    비율 정보 = z / mean(x, y) = 2 * z / (x + y) (43--79)
 9  table    다이아몬드의 가장 넓은 지점에 비해 상단(테이블)의 너비 (43--95)



,price,carat,cut,color,clarity,x,y,z,depth,table
0,326,0.230,Ideal,E,SI2,3.950,3.980,2.430,61.500,55.000
1,326,0.210,Premium,E,SI1,3.890,3.840,2.310,59.800,61.000
2,327,0.230,Good,E,VS1,4.050,4.070,2.310,56.900,65.000
3,334,0.290,Premium,I,VS2,4.200,4.230,2.630,62.400,58.000
4,335,0.310,Good,J,SI2,4.340,4.350,2.750,63.300,58.000


### #02 자료형 점검

1. 자료형 확인

In [3]:
origin.info()

<class 'pandas.DataFrame'>
RangeIndex: 53940 entries, 0 to 53939
Data columns (total 10 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   price    53940 non-null  int64  
 1   carat    53940 non-null  float64
 2   cut      53940 non-null  str    
 3   color    53940 non-null  str    
 4   clarity  53940 non-null  str    
 5   x        53940 non-null  float64
 6   y        53940 non-null  float64
 7   z        53940 non-null  float64
 8   depth    53940 non-null  float64
 9   table    53940 non-null  float64
dtypes: float64(6), int64(1), str(3)
memory usage: 4.7 MB


2. 자료형 변환

In [4]:
df1 = my_qtcheck.set_type(origin, as_category=['cut', 'color', 'clarity'])
df1.head()

<class 'pandas.DataFrame'>
RangeIndex: 53940 entries, 0 to 53939
Data columns (total 10 columns):
 #   Column   Non-Null Count  Dtype   
---  ------   --------------  -----   
 0   price    53940 non-null  int64   
 1   carat    53940 non-null  float64 
 2   cut      53940 non-null  category
 3   color    53940 non-null  category
 4   clarity  53940 non-null  category
 5   x        53940 non-null  float64 
 6   y        53940 non-null  float64 
 7   z        53940 non-null  float64 
 8   depth    53940 non-null  float64 
 9   table    53940 non-null  float64 
dtypes: category(3), float64(6), int64(1)
memory usage: 3.0 MB


,price,carat,cut,color,clarity,x,y,z,depth,table
0,326,0.230,Ideal,E,SI2,3.950,3.980,2.430,61.500,55.000
1,326,0.210,Premium,E,SI1,3.890,3.840,2.310,59.800,61.000
2,327,0.230,Good,E,VS1,4.050,4.070,2.310,56.900,65.000
3,334,0.290,Premium,I,VS2,4.200,4.230,2.630,62.400,58.000
4,335,0.310,Good,J,SI2,4.340,4.350,2.750,63.300,58.000


### #03 데이터 중복 점검

1. 중복 검사 및 중복 데이터 삭제

In [5]:
df2 = my_qtcheck.check_duplicates(df1)
df2.head()

중복된 행의 수: 146
중복된 행이 제거되었습니다.


,price,carat,cut,color,clarity,x,y,z,depth,table
0,326,0.230,Ideal,E,SI2,3.950,3.980,2.430,61.500,55.000
1,326,0.210,Premium,E,SI1,3.890,3.840,2.310,59.800,61.000
2,327,0.230,Good,E,VS1,4.050,4.070,2.310,56.900,65.000
3,334,0.290,Premium,I,VS2,4.200,4.230,2.630,62.400,58.000
4,335,0.310,Good,J,SI2,4.340,4.350,2.750,63.300,58.000


### #04 결측치 점검

1. 결측치 확인

In [6]:
my_qtcheck.check_missing_values(df2)

,Missing Count,Missing Ratio (%)
price,0,0.000
carat,0,0.000
cut,0,0.000
color,0,0.000
clarity,0,0.000
x,0,0.000
y,0,0.000
z,0,0.000
depth,0,0.000
table,0,0.000


### #05 품질 검사 저장

1. 품질 검사 데이터셋 저장

In [7]:
df2.to_excel("diamonds_qtcheck.xlsx", index=False)

### #06 기술 통계량 확인

1. 명목형 변수의 기술 통계량 확인

In [8]:
category_desc = my_qtcheck.categorical_summary(df2, save_path="diamonds_category_summary.xlsx")
category_desc

📊 컬럼 'cut'의 value_counts():


,count
cut,
Fair,1598
Good,4891
Ideal,21488
Premium,13748
Very Good,12069


📊 컬럼 'color'의 value_counts():


,count
color,
D,6755
E,9776
F,9520
G,11262
H,8272
I,5407
J,2802


📊 컬럼 'clarity'의 value_counts():


,count
clarity,
I1,740
IF,1784
SI1,13032
SI2,9150
VS1,8156
VS2,12229
VVS1,3647
VVS2,5056


,cut,color,clarity
count,53794,53794,53794
unique,5,7,8
top,Ideal,G,SI1
freq,21488,11262,13032


2. 연속형 변수의 기술 통계량 확인

In [9]:
desc_df = my_qtcheck.numerical_summary(df2, save_path="diamonds_numerical_summary.xlsx")
desc_df

,count,mean,std,min,25%,50%,75%,max,rel_diff,rdiff_flag,iqr,upper_bound,lower_bound,upper_outliers,upper_outliers_ratio,lower_outliers,lower_outliers_ratio,outliers,outliers_ratio,skew,skew_interpret,kurt,kurt_interpret,log_need
price,53794.000,3933.065,3988.114,326.000,951.000,2401.000,5326.750,18823.000,0.638,large_diff,4375.750,11890.375,-5612.625,3523,0.065,0,0.000,3523,0.065,1.618,right tail,2.178,leptokurtic,log
carat,53794.000,0.798,0.473,0.200,0.400,0.700,1.040,5.010,0.140,diff,0.640,2.000,-0.560,1873,0.035,0,0.000,1873,0.035,1.114,right tail,1.247,leptokurtic,log
x,53794.000,5.731,1.121,0.000,4.710,5.700,6.540,10.740,0.005,similar,1.830,9.285,1.965,24,0.000,7,0.000,31,0.001,0.380,symmetric,-0.629,platykurtic,none
y,53794.000,5.735,1.141,0.000,4.720,5.710,6.540,58.900,0.004,similar,1.820,9.270,1.990,22,0.000,6,0.000,28,0.001,2.446,right tail,91.752,leptokurtic,log1p
z,53794.000,3.539,0.705,0.000,2.910,3.530,4.030,31.800,0.002,similar,1.120,5.710,1.230,28,0.001,20,0.000,48,0.001,1.529,right tail,47.384,leptokurtic,log1p
depth,53794.000,61.748,1.430,43.000,61.000,61.800,62.500,79.000,0.001,similar,1.500,64.750,58.750,1027,0.019,1498,0.028,2525,0.047,-0.114,symmetric,5.413,leptokurtic,none
table,53794.000,57.458,2.234,43.000,56.000,57.000,59.000,95.000,0.008,similar,3.000,63.500,51.500,588,0.011,16,0.000,604,0.011,0.792,right tail,2.775,leptokurtic,log


3. 연속형 변수의 기술 통계량을 전치 형태로 출력

In [10]:
desc_df.T

,price,carat,x,y,z,depth,table
count,53794.000,53794.000,53794.000,53794.000,53794.000,53794.000,53794.000
mean,3933.065,0.798,5.731,5.735,3.539,61.748,57.458
std,3988.114,0.473,1.121,1.141,0.705,1.430,2.234
min,326.000,0.200,0.000,0.000,0.000,43.000,43.000
25%,951.000,0.400,4.710,4.720,2.910,61.000,56.000
50%,2401.000,0.700,5.700,5.710,3.530,61.800,57.000
75%,5326.750,1.040,6.540,6.540,4.030,62.500,59.000
max,18823.000,5.010,10.740,58.900,31.800,79.000,95.000
rel_diff,0.638,0.140,0.005,0.004,0.002,0.001,0.008
rdiff_flag,large_diff,diff,similar,similar,similar,similar,similar


## 연습문제 - Housing Prices Dataset (yasserh)

In [11]:
# 1. 라이브러리 참조
from jussam import load_data
from helpers import my_qtcheck

In [14]:
# 2. 데이터 불러오기
prices = load_data("housing")
prices.head()

📚 주택 가격 데이터셋. 주택 면적, 침실 수, 가구 비치 여부, 주요 도로와의 근접성 등 특정 요소를 기반으로 주택 가격을 예측하는 간단하면서도 도전적인 프로젝트입니다. 데이터셋은 작지만, 다중공선성이 매우 높아 예측 모델 구축에 어려움이 있습니다. 이러한 난관을 극복하고 괜찮은 예측 모델을 만들 수 있을까요? (출처: https://www.kaggle.com/datasets/yasserh/housing-prices-dataset)

    field             description
--  ----------------  ----------------------------------------------------------------------------------------------------------------------------------------------------------------
 0  price             주택 가격 (USD 달러) - 예측 대상 변수
 1  area              주택 면적 (제곱 단위) - 주택의 건축 면적
 2  bedrooms          침실 수 (개수) - 주택의 침실 개수
 3  bathrooms         욕실 수 (개수) - 주택의 욕실 개수
 4  stories           주택 층 수 (개수) - 주택의 총 지상 층수
 5  mainroad          주요 도로 인접 여부 (이진형: Yes/No 또는 0/1) - 주택이 주요 도로에 인접해 있는지 여부
 6  guestroom         게스트룸 보유 여부 (이진형: Yes/No 또는 0/1) - 손님용 방이 있는지 여부
 7  basement          지하실 보유 여부 (이진형: Yes/No 또는 0/1) - 지하실이 있는지 여부
 8  hotwaterheating   온수 난방 시스템 보유 여부 (이진형: Yes/No 또는 0/1) - 온수 난방 시스템이 설치되어 있는지 여부
 9  a

,price,area,bedrooms,bathrooms,stories,mainroad,guestroom,basement,hotwaterheating,airconditioning,parking,prefarea,furnishingstatus
0,13300000,7420,4,2,3,yes,no,no,no,yes,2,yes,furnished
1,12250000,8960,4,4,4,yes,no,no,no,yes,3,no,furnished
2,12250000,9960,3,2,2,yes,no,yes,no,no,2,yes,semi-furnished
3,12215000,7500,4,2,2,yes,no,yes,no,yes,3,yes,furnished
4,11410000,7420,4,1,2,yes,yes,yes,no,yes,2,no,furnished


In [15]:
# 3. 자료형 확인
prices.info()

<class 'pandas.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   price             545 non-null    int64
 1   area              545 non-null    int64
 2   bedrooms          545 non-null    int64
 3   bathrooms         545 non-null    int64
 4   stories           545 non-null    int64
 5   mainroad          545 non-null    str  
 6   guestroom         545 non-null    str  
 7   basement          545 non-null    str  
 8   hotwaterheating   545 non-null    str  
 9   airconditioning   545 non-null    str  
 10  parking           545 non-null    int64
 11  prefarea          545 non-null    str  
 12  furnishingstatus  545 non-null    str  
dtypes: int64(6), str(7)
memory usage: 69.2 KB


In [17]:
# 3-1. 자료형 변환
df1 = my_qtcheck.set_type(prices, as_category=['mainroad', 'guestroom', 'basement', 'hotwaterheating', 'airconditioning', 'prefarea', 'furnishingstatus'])
df1.head()

<class 'pandas.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype   
---  ------            --------------  -----   
 0   price             545 non-null    int64   
 1   area              545 non-null    int64   
 2   bedrooms          545 non-null    int64   
 3   bathrooms         545 non-null    int64   
 4   stories           545 non-null    int64   
 5   mainroad          545 non-null    category
 6   guestroom         545 non-null    category
 7   basement          545 non-null    category
 8   hotwaterheating   545 non-null    category
 9   airconditioning   545 non-null    category
 10  parking           545 non-null    int64   
 11  prefarea          545 non-null    category
 12  furnishingstatus  545 non-null    category
dtypes: category(7), int64(6)
memory usage: 29.6 KB


,price,area,bedrooms,bathrooms,stories,mainroad,guestroom,basement,hotwaterheating,airconditioning,parking,prefarea,furnishingstatus
0,13300000,7420,4,2,3,yes,no,no,no,yes,2,yes,furnished
1,12250000,8960,4,4,4,yes,no,no,no,yes,3,no,furnished
2,12250000,9960,3,2,2,yes,no,yes,no,no,2,yes,semi-furnished
3,12215000,7500,4,2,2,yes,no,yes,no,yes,3,yes,furnished
4,11410000,7420,4,1,2,yes,yes,yes,no,yes,2,no,furnished


In [18]:
# 4. 데이터 중복 점검
df2 = my_qtcheck.check_duplicates(df1)
df2.head()

중복된 행의 수: 0


,price,area,bedrooms,bathrooms,stories,mainroad,guestroom,basement,hotwaterheating,airconditioning,parking,prefarea,furnishingstatus
0,13300000,7420,4,2,3,yes,no,no,no,yes,2,yes,furnished
1,12250000,8960,4,4,4,yes,no,no,no,yes,3,no,furnished
2,12250000,9960,3,2,2,yes,no,yes,no,no,2,yes,semi-furnished
3,12215000,7500,4,2,2,yes,no,yes,no,yes,3,yes,furnished
4,11410000,7420,4,1,2,yes,yes,yes,no,yes,2,no,furnished


In [19]:
# 5. 결측치 점검
my_qtcheck.check_missing_values(df2)

,Missing Count,Missing Ratio (%)
price,0,0.000
area,0,0.000
bedrooms,0,0.000
bathrooms,0,0.000
stories,0,0.000
mainroad,0,0.000
guestroom,0,0.000
basement,0,0.000
hotwaterheating,0,0.000
airconditioning,0,0.000


In [ ]:
# 6. 품질 검사 데이터셋 저장
df2.to_excel("Housing_qtcheck.xlsx", index=False)

In [21]:
# 7. 기술 통계량 확인 - 명목형
category_desc = my_qtcheck.categorical_summary(df2, save_path="Housing_category_summary.xlsx")
category_desc

📊 컬럼 'mainroad'의 value_counts():


,count
mainroad,
no,77
yes,468


📊 컬럼 'guestroom'의 value_counts():


,count
guestroom,
no,448
yes,97


📊 컬럼 'basement'의 value_counts():


,count
basement,
no,354
yes,191


📊 컬럼 'hotwaterheating'의 value_counts():


,count
hotwaterheating,
no,520
yes,25


📊 컬럼 'airconditioning'의 value_counts():


,count
airconditioning,
no,373
yes,172


📊 컬럼 'prefarea'의 value_counts():


,count
prefarea,
no,417
yes,128


📊 컬럼 'furnishingstatus'의 value_counts():


,count
furnishingstatus,
furnished,140
semi-furnished,227
unfurnished,178


,mainroad,guestroom,basement,hotwaterheating,airconditioning,prefarea,furnishingstatus
count,545,545,545,545,545,545,545
unique,2,2,2,2,2,2,3
top,yes,no,no,no,no,no,semi-furnished
freq,468,448,354,520,373,417,227


In [23]:
# 7. 기술 통계량 확인 - 연속형
desc_df = my_qtcheck.numerical_summary(df2, save_path="Housing_numerical_summary.xlsx")
desc_df.T

,price,area,bedrooms,bathrooms,stories,parking
count,545.000,545.000,545.000,545.000,545.000,545.000
mean,4766729.248,5150.541,2.965,1.286,1.806,0.694
std,1870439.616,2170.141,0.738,0.502,0.867,0.862
min,1750000.000,1650.000,1.000,1.000,1.000,0.000
25%,3430000.000,3600.000,2.000,1.000,1.000,0.000
50%,4340000.000,4600.000,3.000,1.000,2.000,0.000
75%,5740000.000,6360.000,3.000,2.000,2.000,1.000
max,13300000.000,16200.000,6.000,4.000,4.000,3.000
rel_diff,0.098,0.120,0.012,0.286,0.097,inf
rdiff_flag,similar,diff,similar,diff,similar,large_diff


### 인사이트 탐색

#### 1. 전체 형태 파악
관측치 545 건, 모든 변수 동일   
분석 대상 변수 13개   
결측치 0건, count 모두 동일   

- 관측치 545건, 모든 변수의 count가 동일 -> 결측치 없는 완결된 데이터
- 분석 대상은 가격(price) 1개 + 주택 속성 12개 -> 회귀(가격 예측) 문제에 적합
- 변수별 단위, 스케일이 상이 (price 175만~1,330만 vs area 1,650~16,200)

#### 2. 중심 위치 파악
price: 평균 4,766,729 / 중앙값 4,340,000   
area: 평균 5,150 / 중앙값 4,600   
bedrooms: 평균 2.965 / 중앙값 3   
bathrooms: 평균 1.286 / 중앙값 1   
stories: 평균 1.806 / 중앙값 2   
parking: 평균 0.694 / 중앙값 0   

- 왜도, 첨도 진단표   

| 변수 | 왜도(skew) | 꼬리 방향 | 첨도(kurt) | 분포 특성 | 로그 변환 |
|---|---|---|---|---|---|
| **price** | 1.212 | 오른쪽 꼬리 | 1.960 | 뾰족한 분포(leptokurtic) | 권장 |
| **area** | 1.321 | 오른쪽 꼬리 | 2.751 | 뾰족한 분포(leptokurtic) | 권장 |
| **bedrooms** | 0.496 | 거의 대칭 | 0.728 | 뾰족한 분포(leptokurtic) | 불필요 |
| **bathrooms** | 1.589 | 오른쪽 꼬리 | 2.165 | 뾰족한 분포(leptokurtic) | 권장 |
| **stories** | 1.082 | 오른쪽 꼬리 | 0.679 | 뾰족한 분포(leptokurtic) | 권장 |
| **parking** | 0.842 | 오른쪽 꼬리 | -0.573 | 평평한 분포(platykurtic) | 불필요 |

- price, area는 평균 > 중앙값, 왜도 > 1, log 권장
- bathrooms, stories는 평균 ≈ 중앙값, 왜도 > 1, log 권장
- bedrooms,  parking은 평균 ≈ 중앙값, 왜도 < 1, 정규분포에 가까움

#### 3. 변동, 산포 파악

| 변수 | 평균 | 표준편차(std) | CV(약) | 상대적 변동 | 범위 |
|---|---:|---:|---:|---|---:|
| **price** | 4,766,729 | 1,870,440 | 39% | 중간 | 약 7.6배 |
| **area** | 5,151 | 2,170 | 42% | 중간 | 약 9.8배 |
| **bedrooms** | 2.965 | 0.738 | 25% | 비교적 작음 | 6배 |
| **bathrooms** | 1.286 | 0.502 | 39% | 중간 | 4배 |
| **stories** | 1.806 | 0.867 | 48% | 비교적 큼 | 4배 |
| **parking** | 0.694 | 0.862 | 124% | 해석 주의 | 계산 불가 |

- parking의 표준편차(0.862)가 평균(0.694)보다 큼 -> 변동성 극단적

#### 4. 사분위수, IQR로 이상치 경계 진단

| 변수 | Q1 | Q3 | IQR | 하한 | 상한 | 이상치 수 | 비율 |
|---|---:|---:|---:|---:|---:|---:|---:|
| price | 3,430,000 | 5,740,000 | 2,310,000 | -35,000 | 9,205,000 | 15 | 2.8% |
| area | 3,600 | 6,360 | 2,760 | -540 | 10,500 | 12 | 2.2% |
| bedrooms | 2 | 3 | 1 | 0.5 | 4.5 | 12 | 2.2% |
| bathrooms | 1 | 2 | 1 | -0.5 | 3.5 | 1 | 0.2% |
| stories | 1 | 2 | 1 | -0.5 | 3.5 | 41 | 7.5% |
| parking | 0 | 1 | 1 | -1.5 | 2.5 | 12 | 2.2% |

- stories는 이상치가 7.5% 존재 (가장 심각)
- price, area, bedrooms, stories, parking은 이상치 점검 필요
- 하한이 음수인 변수는 실제 데이터가 0 이상 -> 하한 위반 없음

#### 5. 이상치 점검 및 종합 인사이트

| 변수 | 정상 하한(IQR 기준) | 실제 최솟값 | 정상 상한(IQR 기준) | 실제 최댓값 | 전체 범위(min~max) |
|---|---:|---:|---:|---:|---|
| price | -35,000 | 1,750,000 | 9,205,000 | 13,300,000 | 1,750,000 ~ 13,300,000 |
| area | -540 | 1,650 | 10,500 | 16,200 | 1,650 ~ 16,200 |
| bedrooms | 0.5 | 1 | 4.5 | 6 | 1 ~ 6 |
| bathrooms | -0.5 | 1 | 3.5 | 4 | 1 ~ 4 |
| stories | -0.5 | 1 | 3.5 | 4 | 1 ~ 4 |
| parking | -1.5 | 0 | 2.5 | 3 | 0 ~ 3 |

- 가격은 전반적으로 오른쪽 꼬리가 길고 일부 고가 주택이 존재하며, 이 값들이 평균을 위쪽으로 끌어올리고 있을 가능성이 있다.
- bedrooms,  parking은 평균 ≈ 중앙값, 표준편차 < 1이므로 매우 안정적이다. 가격 예측 모델에서 기여도가 낮을 가능성이 있다.

## 💡 인사이트 (모범답안)

### Q1. 수치/범주 비율은?

- 전체 변수는 **13개**이며, **수치형 6개(46.2%)**, **범주형 7개(53.8%)** 이다.
- 수치형: `price`, `area`, `bedrooms`, `bathrooms`, `stories`, `parking`
- 범주형: `mainroad`, `guestroom`, `basement`, `hotwaterheating`, `airconditioning`, `prefarea`, `furnishingstatus`
- `yes/no` 컬럼은 의미상 수치가 아니라 **이진 범주형 변수**이므로, 품질 점검 단계에서는 **`category` 타입으로 변환**하는 것이 적절하다.
- 이후 회귀/머신러닝 모델링 단계에서는 필요에 따라 **`yes=1, no=0`** 또는 더미변수로 인코딩하면 된다.

### Q2. 결측·중복 있나?

- `info()`와 결측치 점검 결과, **모든 컬럼의 결측치는 0건**이다.
- `duplicated()` 점검 결과, **중복 행은 0건**이다.
- 따라서 이 데이터셋은 **결측치 처리와 중복 제거가 필요하지 않다.**

### Q3. `price` 분포 형태는?

- `price`의 평균은 **4,766,729**, 중앙값은 **4,340,000**으로 평균이 더 크다.
- `area`의 평균은 **5,150.5**, 중앙값은 **4,600**으로 역시 평균이 더 크다.
- 왜도(skew)는 `price = 1.212`, `area = 1.321`로 둘 다 **우편향(right-skewed)** 분포이다.
- 따라서 **`price`와 `area`는 로그 변환 우선 검토 대상**이다.
- 추가로 모듈 기준에서는 `bathrooms`, `stories`도 로그 변환 후보로 판정되지만, 핵심 대상 변수는 **`price`와 `area`**이다.

### Q4. 이상치 경계는?

- `price`
  - 하한 = **-35,000**
  - 상한 = **9,205,000**
  - 상한 초과 이상치 = **15건 (2.75%)**
  - 하한 미만 이상치 = **0건 (0.00%)**

- `area`
  - 하한 = **-540**
  - 상한 = **10,500**
  - 상한 초과 이상치 = **12건 (2.20%)**
  - 하한 미만 이상치 = **0건 (0.00%)**

- `bedrooms`
  - 하한 = **0.5**
  - 상한 = **4.5**
  - 상한 초과 이상치 = **12건 (2.20%)**
  - 하한 미만 이상치 = **0건 (0.00%)**

- `bathrooms`
  - 하한 = **-0.5**
  - 상한 = **3.5**
  - 상한 초과 이상치 = **1건 (0.18%)**
  - 하한 미만 이상치 = **0건 (0.00%)**

- `stories`
  - 하한 = **-0.5**
  - 상한 = **3.5**
  - 상한 초과 이상치 = **41건 (7.52%)**
  - 하한 미만 이상치 = **0건 (0.00%)**

- `parking`
  - 하한 = **-1.5**
  - 상한 = **2.5**
  - 상한 초과 이상치 = **12건 (2.20%)**
  - 하한 미만 이상치 = **0건 (0.00%)**


### 질문 내용을 통해 도출한 결론

#### Step 1·2

- `price`와 `area`는 평균 > 중앙값, 왜도 > 1로 확인되어 **우편향 분포**이다.
- 따라서 두 변수는 **로그 변환 우선 대상**으로 식별된다.

#### Step 3·4

- 변동계수(CV)는 `area = 0.421`, `price = 0.392`로, **`area`의 상대 변동성이 약간 더 크다.**
- IQR 기준 상한 초과 비율은 `area 2.20%`, `price 2.75%`로 확인되었다.

#### Step 5

- 범주형 시설 변수는 가격 프리미엄과 관련이 있다.
  - `airconditioning = yes`는 `no` 대비 평균 가격이 **약 43.45% 높다.**
  - `prefarea = yes`는 `no` 대비 평균 가격이 **약 32.85% 높다.**
  - `hotwaterheating = yes`는 `no` 대비 평균 가격이 **약 17.58% 높다.**
  - `furnishingstatus`는 **furnished > semi-furnished > unfurnished** 순으로 평균 가격이 높다.
- 즉, **시설 보유 여부와 선호지역 여부는 가격 프리미엄 형성에 유의미한 영향을 주는 변수**로 볼 수 있다.